stdlib/public/Observation/Sources/Observation/Locking.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+
+@_silgen_name("_swift_observation_lock_size")
+func _lockSize() -> Int
+
+@_silgen_name("_swift_observation_lock_init")
+func _lockInit(_: UnsafeRawPointer)
+
+@_silgen_name("_swift_observation_lock_lock")
+func _lockLock(_: UnsafeRawPointer)
+
+@_silgen_name("_swift_observation_lock_unlock")
+func _lockUnlock(_: UnsafeRawPointer)
+
+@available(SwiftStdlib 5.9, *)
+struct ManagedCriticalState<State> {
+  final private class LockedBuffer: ManagedBuffer<State, UInt8> { }
+
+  private let buffer: ManagedBuffer<State, UInt8>
+
+  init(_ buffer: ManagedBuffer<State, UInt8>) {
+    self.buffer = buffer
+  }
+  
+  init(_ initial: State) {
+    self.init(LockedBuffer.create(minimumCapacity: _lockSize()) { buffer in
```

Comment:
This does not technically guarantee that the managed buffer's storage will have the right alignment to store a lock.